js/actions/windowActions.js

```diff
@@ -333,6 +334,33 @@ const windowActions = {
   },
 
   /**
+   * Dispatches a message to close multiple frames
+   * @param {Object[]} framePropsList - The properties of all frames to close
+   */
+  closeFrames: function (framePropsList) {
+    const activeFrameKey = windowStore.getState().get('activeFrameKey')
```

Comment:
ideally there should be no logic inside the window action, I think you should just make it an easy api and have it declarative, so like windowAction.tabPageClosed(i) for the index to close.  The store then does the work to figure out which frames it needs to close and closes them all in one swoop.